enterprise/frontend/src/embedding-sdk/cli/steps/setup-license.ts

```diff
@@ -0,0 +1,77 @@
+import { input } from "@inquirer/prompts";
+import chalk from "chalk";
+import toggle from "inquirer-toggle";
+import open from "open";
+
+import type { CliStepMethod } from "embedding-sdk/cli/types/cli";
+
+import { printEmptyLines, printInfo, printWithPadding } from "../utils/print";
+import { propagateErrorResponse } from "../utils/propagate-error-response";
+
+const trialUrl = `https://store.metabase.com/checkout?plan=pro&deployment=self-hosted`;
+const trialUrlWithUtm = `${trialUrl}&utm_source=product&utm_medium=checkout&utm_campaign=embedding-sdk&utm_content=embedding-sdk-cli`;
+
+export const setupLicense: CliStepMethod = async state => {
+  const hasLicenseKey = await toggle({
+    message: "Do you already have a Metabase Pro license key?",
+    default: false,
+  });
+
+  if (!hasLicenseKey) {
+    printEmptyLines(1);
+    printWithPadding(
+      `Please sign up for a free trial of Metabase Pro self-hosted or purchase a license.`,
+    );
+
+    const shouldOpenInBrowser = await toggle({
+      message: `Open the store to get a license key?`,
+      default: true,
+    });
+
+    if (shouldOpenInBrowser) {
+      try {
+        await open(trialUrlWithUtm);
+        printWithPadding(`Opened ${chalk.blue(trialUrl)} in your browser.`);
+      } catch (error) {
+        printInfo(`Please visit ${chalk.blue(trialUrl)} to get a license key.`);
+      }
+    }
+  }
+
+  // Activate the license
+  // eslint-disable-next-line no-constant-condition -- ask until user provides a valid license key
+  while (true) {
```

Comment:
should we be doing things like this? I feel like if I'm stuck in a CLI I'd rather just have the CLI exit with an explanation of what I should do to make sure I don't get a problem in the future, rather than looping infinitely

Reply:
@oisincoveney I think it's pretty common to mistype a character in the license, so it'll be pretty bad if we quit the entire flow just because we mistyped the license key - especially when they've already done a lot of setup. This gives them a clear feedback that the license was wrong, and gives them the ability to try again.

Reply:
yeah, that's fair. I think in any case, we should give the user an easy exit somewhere instead of forcing them to `ctrl + c`

Reply:
@oisincoveney I see, I think we could ask them if they want to try again or abandon the flow after a failed license attempt?

Reply:
yeah that seems like a good option 😄

Reply:
[Fixed in this commit](https://github.com/metabase/metabase/pull/46810/commits/d050b1d9ff6da03ccf68be8e3624f7e776ce08d5), thanks!

Reply:
@oisincoveney Instead of failing the flow, I added an option to "activate the license later", as they could've configure the license themselves in the web UI. (It's no longer set via an env var, so it can be set manually later)

![CleanShot 2567-08-21 at 16 50 30@2x](https://github.com/user-attachments/assets/cd94d7db-02f3-4d51-b3bc-dafb2f8ddd75)